qutip/solver/flimesolve.py

```diff
@@ -59,107 +59,103 @@ def _floquet_rate_matrix(floquet_basis,
     '''
     Hdim = len(floquet_basis.e_quasi)
 
-    time = T
+    time = floquet_basis.T
     dt = time / Nt
     tlist = np.linspace(0, time - dt, Nt)
+    omega = 2 * np.pi / time
 
     '''
     First,divide all quasienergies by omega to get everything in terms of
         omega.
     '''
 
-    def delta(a, ap, b, bp, l, lp):
-        return ((floquet_basis.e_quasi[a] - floquet_basis.e_quasi[ap])
-                - (floquet_basis.e_quasi[b] - floquet_basis.e_quasi[bp])) \
-            / (list(omega.values())[0]) \
-            + (l - lp)
+    delta_m = np.add.outer(floquet_basis.e_quasi, -floquet_basis.e_quasi)
+    delta_m = np.add.outer(delta_m, -delta_m)
+    delta_m /= omega
 
-    total_R_tensor = {}
+    total_R_tensor = defaultdict(lambda : 0)
     for cdx, c_op in enumerate(c_ops):
         '''
         These lines transform the lowering operator into the Floquet mode
             basis
         '''
-        fmodes = np.stack([np.stack(
-            [i.full() for i in floquet_basis.mode(t)])
-            for t in tlist])[..., 0]
-        fmodes_ct = np.transpose(fmodes, (0, 2, 1)).conj()
-        c_op_Floquet_basis = (fmodes_ct @ c_op.full() @ fmodes)
+
+        fmodes = [floquet_basis.mode(t, data=True).transpose() for t in tlist]
```

Comment:
`floquet_basis.mode` compute as 2 matrix and split the columns, unless you use `data=True`. Using that options, we don't have to split and rejoin the matrices. 
Also the modes are split in column, but you stack them as rows... Are you sure this transposition is desired?

Reply:
I rewrote this to just use QuTiP's built in '.to_floquet_basis()' transformation. I think the one I had was slightly faster due to vectorization, but unless the user is dealing with a _large_ number of modes I don't think it'll be a huge difference.